routing/cross_mwm_index_graph.hpp

```diff
@@ -97,15 +113,26 @@ class CrossMwmIndexGraph final
     if (!mapping->IsValid())
       return false; // mwm was not loaded.
 
-    MappingGuard mappingGuard(mapping);
-    mapping->LoadCrossContext();
+    auto const it = m_mappingGuards.find(numMwmId);
+    if (it == m_mappingGuards.cend())
+    {
+      m_mappingGuards[numMwmId] = make_unique<MappingGuard>(mapping);
+      mapping->LoadCrossContext();
+    }
+
     fn(numMwmId, mapping);
```

Comment:
BTW, do we really need to pass mwm id to the functor? I guess there's some kind of redundancy:
```c++
LoadWith(mwmId, [&](mwmId, ...) {})
```